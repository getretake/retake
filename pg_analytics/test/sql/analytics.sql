-- Total revenue by date
SELECT event_date, SUM(revenue) AS total_revenue
FROM analytics_test
GROUP BY event_date
ORDER BY event_date;

-- Average session duration
SELECT AVG(session_duration) AS average_session_duration
FROM analytics_test;

-- Count of events by event name
SELECT event_name, COUNT(*) AS event_count
FROM analytics_test
GROUP BY event_name
ORDER BY event_count DESC;

-- Maximum page views by user
SELECT user_id, MAX(page_views) AS max_page_views
FROM analytics_test
GROUP BY user_id;

-- Total revenue for a specific event
SELECT SUM(revenue) AS total_revenue
FROM analytics_test
WHERE event_name = 'Purchase';
