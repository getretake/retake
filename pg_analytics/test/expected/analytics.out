-- Total revenue by date
SELECT event_date, SUM(revenue) AS total_revenue
FROM analytics_test
GROUP BY event_date
ORDER BY event_date;
 event_date | total_revenue 
------------+---------------
 01-01-2024 |         20.00
 01-02-2024 |        150.50
 01-03-2024 |          0.00
 01-04-2024 |          0.00
 01-05-2024 |         30.75
 01-06-2024 |         75.00
 01-07-2024 |          0.00
 01-08-2024 |        200.25
 01-09-2024 |        300.00
 01-10-2024 |         50.00
 01-11-2024 |          0.00
 01-12-2024 |        125.30
 01-13-2024 |          0.00
 01-14-2024 |          0.00
 01-15-2024 |         45.00
 01-16-2024 |         80.00
 01-17-2024 |          0.00
 01-18-2024 |        175.50
 01-19-2024 |        250.00
 01-20-2024 |         60.00
(20 rows)

-- Average session duration
SELECT AVG(session_duration) AS average_session_duration
FROM analytics_test;
 average_session_duration 
--------------------------
     381.5000000000000000
(1 row)

-- Count of events by event name
SELECT event_name, COUNT(*) AS event_count
FROM analytics_test
GROUP BY event_name
ORDER BY event_count DESC;
   event_name   | event_count 
----------------+-------------
 Payment        |           2
 Purchase       |           2
 Login          |           2
 ViewProduct    |           2
 Checkout       |           2
 RemoveFromCart |           2
 AddToCart      |           2
 Signup         |           2
 Logout         |           2
 Review         |           2
(10 rows)

-- Maximum page views by user
SELECT user_id, MAX(page_views) AS max_page_views
FROM analytics_test
GROUP BY user_id;
 user_id | max_page_views 
---------+----------------
       9 |             12
       3 |              7
       5 |              9
       4 |             11
      10 |             10
       6 |              9
       2 |              8
       7 |              7
       1 |             10
       8 |              6
(10 rows)

-- Total revenue for a specific event
SELECT SUM(revenue) AS total_revenue
FROM analytics_test
WHERE event_name = 'Purchase';
 total_revenue 
---------------
        275.80
(1 row)

